### ISSUE TITLE
[bug] `.env` file specified in `envFile` option is cached

### ISSUE THREAD

**Issue by shellscape** [moonrepo/moon#896]
**Describe the bug**

It appears that a `.env` file specified in the `envFile` options is being cached. 

**Steps to reproduce**

Create a `.env` file with:

```
export BATMAN=robin
```

Create a task:

```yml
  batman:
    command: ts-node ./scripts/echo.ts
    options:
      cache: false
      envFile: '/.env'
```

Create `scripts/echo.ts`:

```ts
console.log(process.env.BATMAN);
```

- Run `moon run root:batman`
- Output is: `robin`
- Change the value in the .env file to `joker`
- Run `moon run root:batman -u`
- Output is: `robin`
- Run `moon run root:batman --cache off`
- Output is: `joker`

I'm able to consistently reproduce this in a fresh repo as well as the existing repo.

**Expected behavior**

`.env` files should not be cached, nor considered part of any task cache.

What's also troubling is that the `cache: false` option is specified, but appears to be ignored. `-u` has no effect but `--cache off` does have an affect.

**Screenshots**

n/a

**Environment**

```
MacOS Ventura
Moon v1.7.2
```

**Additional context**

This is something new since migrating from 1.2 to 1.7.x, so this looks like a regression or a breaking change in a minor. 

As an aside, it might be helpful to spin up triage environments that people can fork for reproductions. This is what we setup for rollup https://stackblitz.com/fork/rollup-repro and https://repl.it/@rollup/rollup-plugin-repro

**Comment by milesj** [comment]
@shellscape If you use `moon task <target>`, does the env file path show up as an input?

**Comment by shellscape** [comment]
Sorry for the delay, was out of town for a bit. This is on my list to test on 1.8 later today.

**Comment by milesj** [comment]
This was reworked again in 1.9, can you give it a look.

**Comment by shellscape** [comment]
On it.

**Comment by shellscape** [comment]
@milesj this is still a problem with 1.9.1

**Comment by milesj** [comment]
Ok dug into it more. It's because tasks are expanded when added to the project graph, and then the graph is cached entirely. So subsequent runs are hitting the cache.

This'll take some time to fix.

**Comment by milesj** [comment]
Fixed in v1.11 https://moonrepo.dev/blog/moon-v1.11

**Comment by shellscape** [comment]
DOPE.
